### PR TITLE
test: Split integration and E2E tests for AppStream/NonAppStream

### PR DESCRIPTION
**Pull request by nguyen102**
Issue #, if available:

Description of changes:
* Added three subfolders for` end-to-end-tests` and `integration-tests`. The folders are: `appstream-disabled`, `appstream-enabled`, and `common`.
   * AppStream enabled tests will run all tests in `common` as well as tests in `appstream-enabled`
   * AppStream disabled tests will run all tests in `common` as well as tests in `appstream-disabled`
* I've moved the tests to their appropriate location. 
   * End to end workspace tests were moved to `appstream-disabled` as that current test is testing the AppStream disabled experience. We'll be writing new workspace test for `appstream-enabled`
   * For integration tests, studies and data-sources tests were move to `appstream-disabled` because those tests are [creating Studies](https://github.com/awslabs/service-workbench-on-aws/blob/b799b5bfb1305c07b04742725697c6b0e5f47bdc/main/integration-tests/support/resources/data-sources/studies.js#L50) with `accessType=readWrite` which is not allowed by AppStream [registerStudy code](https://github.com/awslabs/service-workbench-on-aws/blob/64c67609961295a343f5b90a859ff2046ee2c44e/addons/addon-base-raas/packages/base-raas-services/lib/study/study-service.js#L199)
* `package.json` scripts were update to include an option for running `appstream-enabled` tests.
* `run-integration-tests.sh` was updated to include an option for running `appstream-enabled` tests.

To review the changes, click the file filter and only include `.json`, `.md`, and `.sh`. The `.js` files are just files being moved into the correct folder

**Testing:** Check that the script for running tests ran the correct types of test (AppStream Enabled tests or AppStream disabled tests)

Checklist:
- [x] Have you successfully deployed to an AWS account with your changes?
- [x] Have you successfully tested with your changes locally?

AS review ticket id:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: main/integration-tests/__test__/api-tests/common/data-sources/register-ds-study.test.js
Comment (jn1119) [inline main/integration-tests/__test__/api-tests/common/data-sources/register-ds-study.test.js:19]: We seem to have the same filename `register-ds-study.test.js` in both common and non-appstream path. I that expected? I didn't go through the contents in detail but it does seem like they are different.

Comment (nguyen102) [reply main/integration-tests/__test__/api-tests/common/data-sources/register-ds-study.test.js:19]: Ah yes. Each of the two files is a subset of the original `register-ds-study.test.js`. The test cases that passes for the AppStream enabled path is in the common folder. The test cases that doesn't pass for AppStream enabled is in the `appstream-disabled` folder.

Comment (jn1119) [reply main/integration-tests/__test__/api-tests/common/data-sources/register-ds-study.test.js:19]: I see, I wonder why `should return study registration information if admin` doesn't work for AppStream then? That doesn't seem intentional, right? Perhaps we should move it to common cases and open a separate ticket for debugging if it isn't intentional?

Comment (nguyen102) [reply main/integration-tests/__test__/api-tests/common/data-sources/register-ds-study.test.js:19]: `should return study registration information if admin`fails because it trying to [create Studies](https://github.com/awslabs/service-workbench-on-aws/blob/b799b5bfb1305c07b04742725697c6b0e5f47bdc/main/integration-tests/support/resources/data-sources/studies.js#L50) with `accessType=readWrite` which is not allowed by AppStream [registerStudy code](https://github.com/awslabs/service-workbench-on-aws/blob/64c67609961295a343f5b90a859ff2046ee2c44e/addons/addon-base-raas/packages/base-raas-services/lib/study/study-service.js#L199). We don't allow that because if a user can create a study in which they have write access then they'll be able to export data without using the `EgressDataStore`.

Comment (jn1119) [reply main/integration-tests/__test__/api-tests/common/data-sources/register-ds-study.test.js:19]: Ah, so it is an egress store setting as opposed to AppStream setting. I guess we should clarify that with folder structure then. Something like `appstream-and-egress-enabled` perhaps?

Comment (nguyen102) [reply main/integration-tests/__test__/api-tests/common/data-sources/register-ds-study.test.js:19]: good point, I'll change the folder name to reflect that. I'll use `appstream-egress-enabled` to keep the name shorter

File: main/integration-tests/README.md
Comment (jn1119) [inline main/integration-tests/README.md]: Should we include egress in the stage- `AppStreamEgressEnabled`?

File: scripts/run-integration-tests.sh
Comment (jn1119) [inline scripts/run-integration-tests.sh]: Should parameter key and value also reflect egress?

File: scripts/run-integration-tests.sh
Comment (jn1119) [inline scripts/run-integration-tests.sh]: Mention Egress too?

File: main/integration-tests/README.md
Comment (nguyen102) [reply main/integration-tests/README.md]: updated

File: scripts/run-integration-tests.sh
Comment (nguyen102) [reply scripts/run-integration-tests.sh]: updated

File: scripts/run-integration-tests.sh
Comment (nguyen102) [reply scripts/run-integration-tests.sh]: updated

File: scripts/run-integration-tests.sh
Comment (jn1119) [inline scripts/run-integration-tests.sh]: This flag should be `APPSTREAM_EGRESS_ENABLED`, right?

Comment (nguyen102) [reply scripts/run-integration-tests.sh]: sorry about that, good catch. Updated.

**Review by jn1119 — APPROVED** [review]
LGTM. I guess just make sure you run the tests again once before pushing this out.

**Comment by nguyen102** [comment]
Just tested again and all tests passes. I'll wait till after the demo on Friday to merge this in.

**Pull request merged**
